public/app/features/explore/spec/helper/setup.tsx

```diff
@@ -137,16 +140,22 @@ function makeDatasourceSetup({ name = 'loki', id = 1 }: { name?: string; id?: nu
       name: name,
       uid: name,
       query: jest.fn(),
-      getRef: jest.fn(),
+      getRef: jest.fn().mockReturnValue(name),
       meta,
     } as any,
   };
 }
 
-export const waitForExplore = async () => {
-  await screen.findByText(/Editor/i);
+export const waitForExplore = async (exploreId: ExploreId = ExploreId.left) => {
+  const container = await screen.findAllByTestId('data-testid Explore');
+  return await within(container[exploreId === ExploreId.left ? 0 : 1]).findByText(/Editor/i);
 };
```

Comment:
would this work as well?

```suggestion
export const waitForExplore = (exploreId: ExploreId = ExploreId.left) => {
  return withinExplore(exploreId).findByText(/Editor/i);
};
```

Reply:
Yes! Updated in aee63f5591dca40bc6620197702ae0c0e6c444a5